PayloadPkg/FirmwareUpdate/GetCapsuleImage.c

```diff
@@ -337,64 +357,45 @@ GetOsImageList (
   header. It could be read from EMMC, UFS, USB, SATA, etc. block device. Often the
   Capsule image could be saved in the root directory of a FAT system.
 
-  @param[out] FwBuffer        The firmware update capsule image.
-  @param[out] FwSize          The capsule image size.
+  @param[out] CapsuleImage      The firmware update capsule image.
+  @param[out] CapsuleImageSize  The capsule image size.
 
   @retval  EFI_SUCCESS        Get the capsule image successfully.
   @retval  others             Error happening when getting capsule image.
 **/
 EFI_STATUS
 EFIAPI
 GetCapsuleImage (
-  OUT  VOID                      **FwBuffer,
-  OUT  UINT32                    *FwSize
+  OUT VOID     **CapsuleImage,
+  OUT UINT32   *CapsuleImageSize
   )
 {
-  EFI_STATUS                     Status;
-  VOID                           *CapsuleImage;
-  UINTN                          CapsuleImageSize;
-  OS_BOOT_OPTION_LIST            *OsImageInfoList;
-  UINT32                         Index;
+  EFI_STATUS              Status;
+  CAPSULE_INFO_CFG_DATA   *CapsuleInfo;
 
   Status = EFI_UNSUPPORTED;
+  CapsuleInfo = NULL;
 
   DEBUG ((DEBUG_INFO, "\n=================Read Capsule Image==============\n"));
 
-  Status = PlatformGetCapsuleImage (&CapsuleImage, &CapsuleImageSize);
-  DEBUG((DEBUG_ERROR, " PlatformGetCapsuleImage: Status : %r\n", Status));
-  if ((Status == EFI_SUCCESS) || (Status == EFI_NOT_FOUND)) {
-    *FwSize   = CapsuleImageSize;
-    *FwBuffer = CapsuleImage;
-    return Status;
-  }
-
   //
-  // Get Boot Image Info
+  // Get capsule configuration data
   //
-  OsImageInfoList = GetOsImageList ();
-  if (OsImageInfoList == NULL) {
-    DEBUG ((DEBUG_ERROR, "Could not obtain OS image list\n"));
-    return EFI_NOT_FOUND;
-  }
-
+  CapsuleInfo = (CAPSULE_INFO_CFG_DATA *) FindConfigDataByTag (CDATA_CAPSULE_INFO_TAG);
   //
-  // Load and run Image in order from OsImageInfoList
+  // If we do not find capsule information, return error 
   //
-  for (Index = 0; Index < OsImageInfoList->OsBootOptionCount; Index++) {
-    Status = LoadCapsuleImage (&OsImageInfoList->OsBootOption[Index], &CapsuleImage, &CapsuleImageSize);
-    if (!EFI_ERROR (Status)) {
-      break;
-    }
+  if (CapsuleInfo == NULL) {
+    DEBUG((DEBUG_ERROR, " CapsuleInfo not found \n"));
+    return EFI_NOT_FOUND;
   }
 
-  if (Index == OsImageInfoList->OsBootOptionCount) {
-    DEBUG ((DEBUG_ERROR, "Get firmware update capsule image failure!\n"));
+  Status = LoadCapsuleImage (CapsuleInfo, CapsuleImage, CapsuleImageSize);
```

Comment:
Argument should be verified at top level API.  EX:  check if CapsuleImage is NULL, CapsuleImageSize is NULL, etc.